align/pnr/toplevel.py

```diff
@@ -399,10 +400,86 @@ def gen_leaf_bbox_and_hovertext( ctn, p):
                     else:
                         logger.error( f'LEF for concrete name {ctn} (of {atn}) missing.')
 
+
+        tagged_bboxes = defaultdict(dict)
+
+        def per_placement( placement_verilog_d, concrete_name, hN):
```

Comment:
Refactored this code from inside a loop below into its own functions so it can be called separately.